JS1 Week4 HW

alina-kho opened the pull request.

alina-kho committed:
Adding homework files

Kristoffer88 commented on javascript/javascript1/week4/voice-assistant.js:
```diff
@@ -0,0 +1,189 @@
+"use strict";
+
+//Creating global scope variables
+let userName = "";
```
Is there any reason to make this a global scope?

alina-kho replied:
Thanks a lot for the feedback. I fixed variable declaration (let -> const) in the cases you mentioned - everything works perfectly. I also deleted breaks in the switch statement as you said - I don't need them if I have returns. 

As for userName as a global scope, I tried moving it inside the function, but some of the features (1 - when a user says his name again and the assistant answers with "Hi again"; 2 - when a user asks for the name) stopped working. Maybe there is a way out of it.

Kristoffer88 commented on javascript/javascript1/week4/voice-assistant.js:
```diff
@@ -0,0 +1,189 @@
+
+//Creating a function
+function getReply(command) {
+  let commandSplitted = command.split(" "); //Splits the command into separate words for easy search through them
```
Could this be a const?

alina-kho replied:
Fixed.

Kristoffer88 commented on javascript/javascript1/week4/voice-assistant.js:
```diff
@@ -0,0 +1,189 @@
+    command.toLowerCase().includes("my name is") ||
+    command.toLowerCase().includes("i am")
+  ) {
+    let name = commandSplitted[commandSplitted.length - 1];
```
This is not reassigned, so could also be const

alina-kho replied:
Fixed.

Kristoffer88 commented on javascript/javascript1/week4/voice-assistant.js:
```diff
@@ -0,0 +1,189 @@
+  //Name actions
+  //.toLowerCase is used to get rid of case sensitivity
+  if (
+    command.toLowerCase().includes("my name is") ||
```
Great technic to use toLowerCase

Kristoffer88 commented on javascript/javascript1/week4/voice-assistant.js:
```diff
@@ -0,0 +1,189 @@
+      return `Nice to meet you ${name}`;
+    }
+  } else if (command.toLowerCase().includes("is my name")) {
+    if (userName == undefined || userName == "") {
+      return "I haven't met you yet. Please introduce yourself";
+    } else {
+      return userName;
+    }
```
This works great, but you could use the "Return Early" technic. (no need for the last else statement)
If you have time, you could take a look at it: https://dev.to/jenniferlynparsons/early-returns-in-javascript-5hfb

alina-kho replied:
Thanks! I will look into it :)

Kristoffer88 commented on javascript/javascript1/week4/voice-assistant.js:
```diff
@@ -0,0 +1,189 @@
+      "November",
+      "December",
+    ]; // array created to change a number value to a word
+    let today = new Date(); //getting today's date
+    let day = today.getDate();
+    let month = months[today.getMonth()]; //today.getMonth() will give us a value from 0 to 11, so we can easily apply it to find a month in the months array
+    let year = today.getFullYear();
```
Could these be consts?

alina-kho replied:
Fixed.

Kristoffer88 commented on javascript/javascript1/week4/voice-assistant.js:
```diff
@@ -0,0 +1,189 @@
+    switch (randomNum) {
+      case 0:
+        return `You are going to have an amazing day!`;
+        break;
```
I do not think break is nessecary, as you are returning right above it.

alina-kho replied:
Fixed

Kristoffer88 reviewed (approved):
Overall really good.

You might want to look into how to avoid nesting the if/else-if/else statement. To make it a little bit more readable. This is a advanced topic, so only if you have time to look into it

alina-kho committed:
Bug fixes

- Changed let to const where it was reasonable
- Deleted breaks in the switch statement since I made returns in each case
